### ISSUE TITLE
To-Do List #1

### ISSUE THREAD

**Issue by cavaunpeu** [cavaunpeu/vanilla-neural-nets#1]
Renaming:
- [x] Move all abstract base classes into their own module
- [x] Prepend those classes that are intended to be private with underscore
- [x] Change `cost` method in loss functions to `loss`

Functionality:
- [x] Implement and test network parameter objects for recurrent neural network
